templates/common/_base/files/etc-networkmanager-dispatcher.d-90-long-hostname.yaml

```diff
@@ -0,0 +1,40 @@
+filesystem: "root"
+mode: 0755
+path: "/etc/NetworkManager/dispatcher.d/90-long-hostname"
+contents:
+  inline: |
+    #!/bin/bash
+    #
+    # On Google Compute Platform (GCP) the hostname may be too long (>63 chars).
+    # During firstboot the hostname is set in the initramfs before NetworkManager
+    # runs.
+    #
+    # This hook is a simple work around: if the host name is longer than 63
+    # characters, then the hostname is truncated at the _first_ dot.
+    #
+    # Additionally, this hook does not break DNS or cluster resolution,
+    # since NetworkManager sets the appropriate /etc/resolv.conf settings.
+
+
+    IF=$1
+    STATUS=$2
+
+    log() { logger --tag "network-manager/$(basename $0)" "${@}"; }
+
+    if [ "${HOSTAME}" == "localhost" ]; then
+        log "hostname is already set"
+        exit 0
+    fi
+
+    if [[ ! "$STATUS" =~ (up|hostname|dhcp4-change|dhcp4-change) ]]; then
+        exit 0
+    fi
+
+    default_host="${DHCP4_HOST_NAME:-$DHCP6_HOST_NAME}"
+    host=$(echo ${default_host} | cut -f1 -d'.')
```

Comment:
Although VERY unlikely the first dot may be over 63 characters, so maybe we should hard truncate at 63 chars or something, since the hostname will likely not make sense anyways

Reply:
Possibly adding `| cut -c -63` or ` | head -c 63` or  at the end to ensure it's no more than 63 characters.

Reply:
Added, and tested.